jscomp/core/js_of_lam_variant.ml

```diff
@@ -84,3 +84,13 @@ let eval_as_int (arg : J.expression) (dispatches : (int * int) list ) : E.t  =
                body = [S.return (E.int (Int32.of_int  r))],
                       false (* FIXME: if true, still print break*)
               }) dispatches))]
+
+let eval_as_unwrap (arg : J.expression) : E.t =
+  if arg == E.undefined then
+    E.undefined
+  else
```

Comment:
1. I am not sure if it is needed, can you find an example?
2. You need make sure in the front end such case does not happen
```ocaml
[ `a of int | `b]
```